src/gui/editorwidgets/qgsdatetimeeditfactory.cpp

```diff
@@ -114,3 +114,18 @@ QMap<const char*, int> QgsDateTimeEditFactory::supportedWidgetTypes()
   map.insert( QgsDateTimeEdit::staticMetaObject.className(), 10 );
   return map;
 }
+
+unsigned int QgsDateTimeEditFactory::fieldScore( const QgsVectorLayer* vl, int fieldIdx ) const
+{
+  const QgsField field = vl->fields().field( fieldIdx );
+  const QVariant::Type type = field.type();
+  const QgsEditorWidgetConfig config = vl->editFormConfig().widgetConfig( field.name() );
+  if ( type == QVariant::DateTime || config.contains( "field_format" ) )
```

Comment:
I don't have a feeling that QgsDateTimeEditWrapper is wired to support QDate or QTime. It looks to me it supports only QString and QDateTime.